viscy/preprocessing/precompute.py

```diff
@@ -0,0 +1,76 @@
+"""Precompute normalization and store a plain C array"""
+
+from pathlib import Path
+from typing import Generator, Literal
+
+import dask.array as da
+from dask.diagnostics import ProgressBar
+from iohub.ngff.nodes import Plate, Position, Well, open_ome_zarr
+
+
+def _normalize_image(
+    image: da.Array,
+    subtrahend: Literal["mean"] | float,
+    divisor: Literal["std"] | tuple[float, float],
+    eps: float = 1e-6,
+) -> da.Array:
+    if subtrahend == "mean" and divisor == "std":
+        subtrahend_value = image.mean()
+        divisor_value = image.std()
+    else:
+        subtrahend_value, div_lo, div_hi = da.percentile(
+            image.flatten(), (subtrahend, *divisor)
+        )
+        divisor_value = div_hi - div_lo
+    divisor_value = min(divisor_value, eps)
+    return (image - subtrahend_value) / divisor_value
+
+
+def _filter_wells(
+    plate: Plate, include_wells: list[str] | None
+) -> Generator[Well, None, None]:
+    for well_name, well in plate.wells():
+        if include_wells is None or well_name in include_wells:
+            yield well
+
+
+def _filter_fovs(
```

Comment:
here we cannot configure the FOVs per well. what this filters all fovs with the same name for all positions right?

Reply:
I am envisioning cases in like Mantis where fov `A/1/0000000` is good for one well but not for `A/2/000000` and we still want to use this fov.

Reply:
This is controlled by `include_wells` in the calling function below.